public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -1,73 +1,47 @@
 import React, { PureComponent, ChangeEvent } from 'react';
 import { FormLabel, Select, Input, Button } from '@grafana/ui';
-import { DataSourcePluginOptionsEditorProps, DataSourceSettings } from '@grafana/data';
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
 import { SelectableValue } from '@grafana/data';
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
 import CloudWatchDatasource from '../datasource';
 import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
+import { CancelablePromise, makePromiseCancelable } from 'app/core/utils/CancelablePromise';
 
-export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData>;
+const authProviderOptions = [
+  { label: 'Access & secret key', value: 'keys' },
+  { label: 'Credentials file', value: 'credentials' },
+  { label: 'ARN', value: 'arn' },
+] as SelectableValue[];
 
-type CloudwatchSettings = DataSourceSettings<CloudWatchJsonData, CloudWatchSecureJsonData>;
+export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
 export interface State {
-  config: CloudwatchSettings;
-  authProviderOptions: SelectableValue[];
   regions: SelectableValue[];
 }
 
 export class ConfigEditor extends PureComponent<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    const { options } = this.props;
-
     this.state = {
-      config: ConfigEditor.defaults(options),
-      authProviderOptions: [
-        { label: 'Access & secret key', value: 'keys' },
-        { label: 'Credentials file', value: 'credentials' },
-        { label: 'ARN', value: 'arn' },
-      ],
       regions: [],
     };
-
-    this.updateDatasource(this.state.config);
   }
 
-  static getDerivedStateFromProps(props: Props, state: State) {
-    return {
-      ...state,
-      config: ConfigEditor.defaults(props.options),
-    };
-  }
-
-  static defaults = (options: any) => {
-    options.jsonData.authType = options.jsonData.authType || 'credentials';
-    options.jsonData.timeField = options.jsonData.timeField || '@timestamp';
-
-    if (!options.hasOwnProperty('secureJsonData')) {
-      options.secureJsonData = {};
-    }
-
-    if (!options.hasOwnProperty('jsonData')) {
-      options.jsonData = {};
-    }
-
-    if (!options.hasOwnProperty('secureJsonFields')) {
-      options.secureJsonFields = {};
-    }
+  loadRegionsPromise: CancelablePromise<any> = null;
 
-    return options;
-  };
-
-  async componentDidMount() {
-    this.loadRegions();
+  componentDidMount() {
+    this.loadRegionsPromise = makePromiseCancelable(this.loadRegions());
+    this.loadRegionsPromise.promise.catch(({ isCanceled }) => {
+      if (isCanceled) {
+        console.warn('Cloud Watch ConfigEditor has unmounted, intialization was canceled');
+      }
+    });
```

Comment:
Maybe consider cancelling the promise on unmount also?
```suggestion
    });
    
    componentWillUnmount() {
    if (this. loadRegionsPromise) {
      this. loadRegionsPromise.cancel();
    }
  }
```